### PR TITLE
Improve release docs

### PR DESCRIPTION
**Pull request by juliangieseke**
**What type of PR is this?**

**What this PR does/ why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] The commit message explains the changes and why are needed.
- [ ] The code builds and passes lint/style checks locally.
- [ ] The relevant subset of integration tests pass locally.
- [ ] The core changes are covered by tests.
- [ ] The documentation is updated where needed.

### CODE REVIEW

File: README.md
Comment (gracedo) [inline README.md:34]: I read over the removed lines and thought about this some more.. I guess the previous version is probably correct - the tag should be off the latest rc tag, not master in case there were changes to master after the rc (that shouldn't go into stable tag).

The step to add is to merge the stable branch back into master after tagging - noting that there will be merge conflicts to resolve due to adding a new revision (let's say revision 50 for example) in the stable branch that also exists on master. While resolving conflicts, we should use stable's version of revision 50, and move master's revision 50 and any others after it, into a new dir that is stable version + 0.0.1 (next patch version). I'll try to suggest this step below

I just checked KBA to compare, and they are grouping patches in the same dir (`1.0.x` dirs instead of `1.0.0` for example). We can consider moving to this as well to be in line with them

File: README.md
Comment (gracedo) [inline README.md:38]: As mentioned in earlier comment, let's keep the previous version of the above steps that were removed, and thus remove this step.

File: README.md
Comment (gracedo) [inline README.md:35]: let's bring this back!

File: README.md
Comment (gracedo) [inline README.md:36]: and let's bring this back too!

File: README.md
Comment (gracedo) [reply README.md:38]: rather, I will repurpose this by adding the step to merge back stable branch :)

```suggestion
1. merge the `stable-*` branch back into master. there will be merge conflicts to resolve due to different versions of the latest revision file existing on both the master and stable branches. resolve the conflicts by:
  1. keeping the stable branch version of that revision file
  1. creating a new directory with the next patch version (e.g. `1.1.1`)
  1. moving master's version of that revision file and any others following it into that newly created directory
```
^ should new dir be next patch version or minor? I used patch since we separate dirs out by patch versions instead of minor (mentioned this in an earlier comment, kba uses `1.1.x` instead of `1.1.1`)

File: README.md
Comment (gracedo) [inline README.md:42]: I included this step in a suggested step above, so we can remove. Also - should this new dir be the next patch version instead of minor version?

File: README.md
Comment (gracedo) [reply README.md:38]: The merge conflict resolution seems a little risky. Ideally I would like to see this out in a PR for more eyes to double check, my only concern is that hitting the merge button on the PR will also auto-delete the stable branch. We _can_ manually "un-delete" the branch but that also seems sketch and I'm sure people will forget to do it. Is there a GH exception we can make to the auto-delete rule to ignore branches prefixed with `stable-` for example?

Comment (samvantran) [reply README.md:38]: Yes we can mark the branch protected (it already is) and it will not be deleted 

Screen Shot 2020-07-17 at 11 24 16 AM

**Comment by shaneutt** [comment]
@juliangieseke and @gracedo let us know in @mesosphere/sig-ksphere-catalog if there's any help needed to move this one forward :vulcan_salute:

**Comment by shaneutt** [comment]
@juliangieseke ping

File: README.md
Comment (gracedo) [inline README.md:55]:
```suggestion
1. merge that `stable-*` branch back to `master` in order to keep all information on master **NOTE: `master` should NEVER be merged into the `stable-*` branch.**
```
Adding note here that we should never merge master back into any stable branches to avoid getting newer revisions on the stable branch which would be used instead of the older patch revisions. We should also discuss and document here -- should the merge of stable back into master happen via direct commit on the master branch? Or do we want to PR this out -- in which case, the PR will be blocked on the "out of date" warning and requires admin privileges to override the block and merge the PR anyways.

Comment (samvantran) [reply README.md:55]: my 2c: i think opening PRs is good practice generally because 1) you get more eyes on it 2) it gets tested. we might run into a situation one day where the stable-x.y branch _cannot_ be merged back into master bc of a breaking change. That would be good to know and share on a PR and thus we'll know when a divergence occurs.

That said, if we are going with PRs we have to make doubly sure that we don't instinctively click "update with latest master" as Github will tell us to or we revert those master-merges in the same PR if we do it by accident. I myself will be more diligent about it going forward.

**Comment by shaneutt** [comment]
@juliangieseke are you able to follow up on this, or do you need to hand it off to another assignee?

File: README.md
Comment (juliangieseke) [reply README.md:39]: tl;dr
removing these lines and only creating the stable branch after everything is done was correct IMHO.

long
generally speaking, everything "new" should happen on master, the only reason we have `stable-*` _maintenance_ branches is that we sometimes need to patch older versions (e.g. releasing `1.1.1` while master already has `1.2.0-beta.x`) - only in these scenarios we actually _need_ to create a `stable-*` branch for the older version (`stable-1.1.x` in this example). Us doing it earlier is to make it a bit easier to work with the repo.

that said, working on unreleased minor versions ((re- and GA releases), we should mainly/only work on master branch and merging any `stable-*` branch back to master must never result in a conflict. if we get a conflict, something went wrong before and we should revert/fix the root cause instead of trying to duct-tape something with a resolved conflict.


I'd also vote for individual folders per patch version as long as we have the revision concept. we want to be able to version (revision) patch versions indepenendtly.

File: README.md
Comment (juliangieseke) [reply README.md:43]: see more detailed comment above.

tl;dr
we dont want to merge back a newly created `stable-*` to `master`, the two branches must be equal anyway. also, there must never be a conflict when trying to merge any `stable-*` branch back to `master`.

File: README.md
Comment (juliangieseke) [reply README.md]: > should the merge of stable back into master happen via direct commit on the master branch? Or do we want to PR this out

there should never be a conflict and it should only be a fast-forward merge, therefore just merging stable into master by commit would be fine for me. if theres a conflict, something went wrong anyway.

@samvantran made a good point, Im worried about this "update with latest master" button too, also theres "delete branch after merge" 🙈

**Comment by juliangieseke** [comment]
@shaneutt sorry, was out for a while 🙈  went though the feedback and added/changed a couple things. cc @gracedo @samvantran

**Comment by juliangieseke** [comment]
given there are changes on how we want to deal with revisions, I'd suggest to get this PR in with the old setup in mind and create a follow-up for the new setup later. (I can handle that, once KBA is updated and I understand how its supposed to work)

**Comment by shaneutt** [comment]
ping @gracedo

**Comment by samvantran** [comment]
@gracedo is going to be on PTO for this week so we can either 1) go with @juliangieseke's suggestion of getting this PR with the old setup in mind and create a new one to follow up or 2) come back to this PR when she's back

File: README.md
Comment (gracedo) [reply README.md:39]: Is it possible that the pre-release we want to release is not the version of the latest master? I think it makes sense that the latest revision in a version (directory) should always be the one that is released so we shouldn't see conflicts in master vs stable, as you mentioned. But would we ever be in a scenario where, prior to releasing a version e.g. 1.1.0, we'd already created a 1.2.0 dir and started developing for a newer version? If we do think that's a possible scenario, then if we were to branch off master/tag the latest master, the 1.2.0 revisions would be present in a 1.1.0 stable branch.

I think it makes sense to work off the rc tags since they are, aptly named, the release candidates that should eventually become the stable release. And would avoid issues if a newer version underwent development prior to the version getting released.

File: README.md
Comment (gracedo) [reply README.md:43]: > there will be merge conflicts to resolve due to different versions of the latest revision file existing on both the master and stable branches

Tbh I can't remember why there would be this merge conflict, we can reject this and revisit if in the next release we run into something like this again

File: README.md
Comment (juliangieseke) [reply README.md:39]: what you are describing possibly can happen if we forgot to branch off, in such case we'd probably need to revert all newer 1.1.0 revisions, branch off the latest tag, re-add the previously reverted revisions, and merge back 1.1.0 maintenance branch in master. 

so essentially reverting a mistake, and doing it properly.

Comment (gracedo) [reply README.md:39]: I still feel like we may run into the situation where we want to start developing the next version before we are ready to release the current version, which might put us in a bad state. But I'm ok just getting this PR in with this version now since this has been sitting around long enough, and if we happen to run into said scenario in the future, we can revisit the process

Comment (gracedo) [reply README.md:39]: @shaneutt @samvantran any final opinions on the matter? I think otherwise we can get this in!

Comment (samvantran) [reply README.md:39]: i think this is good for now. like you said, we can revisit if we run into the above scenario

**Review by samvantran — APPROVED** [review]

**Review by gracedo — APPROVED** [review]

File: README.md
Comment (juliangieseke) [reply README.md:39]: > I still feel like we may run into the situation where we want to start developing the next version before we are ready to release the current version, which might put us in a bad state. 

thats possible, we just need to branch off of master before we create the new versions folder. from that point on we can work on multiple branches (just like we do currently for 1.1. & 1.2). only important bit is that we branch off _before_ creating new version folders. that the only important thing to keep in mind.

**Pull request merged**
